server.go

```diff
@@ -78,11 +78,11 @@ type mongoServerInfo struct {
 
 var defaultServerInfo mongoServerInfo
 
-func newServer(addr string, tcpaddr *net.TCPAddr, sync chan bool, dial dialer, minPoolSize, maxIdleTimeMS int) *mongoServer {
+func newServer(addr string, raddr net.Addr, sync chan bool, dial dialer, minPoolSize, maxIdleTimeMS int) *mongoServer {
 	server := &mongoServer{
 		Addr:          addr,
-		ResolvedAddr:  tcpaddr.String(),
-		tcpaddr:       tcpaddr,
+		ResolvedAddr:  (raddr).String(),
```

Comment:
The `()` around `(raddr)` are redundant but I think `ResolvedAddr` itself is completely redundant - all it's uses can be replaced with `raddr.String()` I think? Not really sure why it was here in the first place...